### PR TITLE
Added Export function with different file formats.

### LINKED ISSUE #551: Export Analytics Data to Local Device (most likely HoloLense)
The analytics data for the currently loaded room that are stored on the backend should be exportable to a local file.

TODO:

- [x] Implement endpoint in backend to export all analytics data for a project to JSON
- [x] Export data to JSON file locally
- [ ] Optional: Add XML export

### PR DESCRIPTION
**Pull request by PascalRaabe**
closes #551

### CODE REVIEW

File: Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/ExportAnalytics.cs [hunk 1]
```diff
@@ -1,42 +1,63 @@
...
+
+                case ExportSelection.CSV:
+                    fileExtension = ".zip";
+                    break;
+
+                default:
+                    fileExtension = ".txt";
+                    break;
+            }
+            Debug.Log("Data is:" + data);
```
Comment (CytexalMika) [inline Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/ExportAnalytics.cs]: Unnecessary log

File: Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        public event EventHandler WindowClosed;
+
+        private bool isConfiguring = true;
+
+        public ExportSelection ChosenExportSelection;
+
+        public IDropdownConfiguration DropdownConfiguration { get; private set; }
+
+        public bool WindowEnabled
+        { // not needed for configuration window => does not have an effect
```
Comment (LennartCode) [inline Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs]: Comment should be on its own line.

File: Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+using i5.VIAProMa.UI.DropdownMenu;
+using i5.VIAProMa.UI.ListView.Core;
+using i5.VIAProMa.UI.ListView.Strings;
+using i5.VIAProMa.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace i5.VIAProMa.Shelves.IssueShelf
```
Comment (LennartCode) [inline Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs]: Probably wrong namespace

File: Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        {
+            if (!ExternallyInitialized)
+            {
+                Initialize();
+            }
+        }
+
+        public void Initialize()
+        {
+            // populate the source dropdown menu with the available data sources
```
Comment (LennartCode) [inline Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs]: Please stick to the C# style guide and capitalize the first letter of a comment. Also add a dot afterwards

File: Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            List<StringData> sources = new List<StringData>();
+            foreach (ExportSelection source in Enum.GetValues(typeof(ExportSelection)))
+            {
+                sources.Add(new StringData(source.GetDescription()));
+            }
+            sourceSelection.Items = sources;
+
+            //sourceSelection.ItemSelected += SourceSelected; Whatever this does??
```
Comment (LennartCode) [inline Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs]: Do not comment out code

File: Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+            SetExportSelection(ExportSelection.JSON); // first entry of dropdown box is Requirements Bazaar, so set this as the default 
+
+            isConfiguring = false;
+        }
+
+        /*
```
Comment (LennartCode) [inline Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/DropdownCapability.cs]: Do not comment out code

File: Frontend/VIAProMa/Assets/Scripts/UI/AdvancedSettings/ExportSelectionDropdownMenu.cs [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+        public event EventHandler<EventArgs> ItemSelected;
+
+        public List<DataType> Items
+        {
+            get { return itemController?.Items; }
+            set
+            {
+                itemController.Items = value;
+                // by standard select the first element
```
Comment (LennartCode) [inline Frontend/VIAProMa/Assets/Scripts/UI/AdvancedSettings/ExportSelectionDropdownMenu.cs:25]: Again, please stick to the style guide for comments.

**Review by LennartCode — CHANGES_REQUESTED** [review]
Please refer to the comments and resolve these problems.

File: Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/IDropdownConfiguration.cs [hunk 8]
```diff
@@ -0,0 +1,11 @@
+using i5.VIAProMa.DataModel.API;
+
+namespace i5.VIAProMa.Shelves.IssueShelf
```
Comment (CytexalMika) [inline Frontend/VIAProMa/Assets/Scripts/Analytics/FileExport/IDropdownConfiguration.cs]: Wrong namespace again

**Review by LennartCode — APPROVED** [review]
Looks great!

**Review by CytexalMika — CHANGES_REQUESTED** [review]

**Review by CytexalMika — APPROVED** [review]
LGTM

**Pull request merged**